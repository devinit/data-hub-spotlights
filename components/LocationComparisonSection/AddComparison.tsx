import React, { FunctionComponent } from 'react';
import { SpotlightBanner, SpotlightBannerAside } from '../SpotlightBanner';
import { PageSection } from '../PageSection';
import { AddLocation } from './AddLocation';

interface AddComparisonProps {
  onAddComparison: (widgetState?: boolean, tagName?: string) => void;
}

const AddComparison: FunctionComponent<AddComparisonProps> = ({ onAddComparison }) => {
  return (
    <PageSection>
      <SpotlightBanner>
        <SpotlightBannerAside>
          <AddLocation active={true} label={'Add another comparison'} onWidgetClick={onAddComparison}></AddLocation>
        </SpotlightBannerAside>
      </SpotlightBanner>
    </PageSection>
  );
};

export { AddComparison };
