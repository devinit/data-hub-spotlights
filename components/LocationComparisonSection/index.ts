export * from './LocationComparisonSection';
export * from './AddLocation';
