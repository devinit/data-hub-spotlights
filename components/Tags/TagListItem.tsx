import React, { FunctionComponent } from 'react';
import { Button } from '../Button';

interface TagsListItemProps {
  label: string;
  onRemove?: (label: string) => void;
}

const TagListItem: FunctionComponent<TagsListItemProps> = ({ label, onRemove: onRemoveTag }) => {
  const onRemove = (): void => {
    if (onRemoveTag) {
      onRemoveTag(label);
    }
  };

  return (
    <li className="m-pills__item">
      {label.toLowerCase()}
      <Button onClick={onRemove} className="">
        <i role="presentation" aria-hidden="true" className="ico ico--20 ico-cross-slate"></i>
      </Button>
      <style jsx>{`
        text-transform: capitalize;
      `}</style>
    </li>
  );
};

export { TagListItem };
