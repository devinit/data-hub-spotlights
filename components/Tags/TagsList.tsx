import React, { FunctionComponent } from 'react';

const TagsList: FunctionComponent = ({ children }) => {
  return <ul className="m-pills">{children}</ul>;
};

export { TagsList };
