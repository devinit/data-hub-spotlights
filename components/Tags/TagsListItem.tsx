import React, { FunctionComponent } from 'react';

interface TagsListItemProps {
  label: string;
  onRemoveTag?: (label: string) => void;
}

const TagsListItem: FunctionComponent<TagsListItemProps> = ({ label, onRemoveTag }) => {
  const onRemove = (): void => {
    onRemoveTag ? onRemoveTag(label) : null;
  };

  return (
    <li className="m-pills__item">
      <style jsx>{`
        .m-pills__item {
          margin-bottom: 5px !important;
        }
      `}</style>
      <button type="button" onClick={onRemove}>
        <i role="presentation" aria-hidden="true" className="ico ico--16 ico-plus-blank"></i>
      </button>
      {label}
    </li>
  );
};

export { TagsListItem };
