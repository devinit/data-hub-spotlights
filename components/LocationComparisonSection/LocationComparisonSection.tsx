import React, { FunctionComponent, useState, ReactNode } from 'react';
import { PageSection, PageSectionHeading } from '../PageSection';
import { Spotlight } from '../Spotlight';
import { SpotlightTheme, SpotlightLocation } from '../../utils';
import { LocationFiltersAndCharts } from './LocationFiltersAndCharts';
import { AddComparison } from './AddComparison';
import { LocationComparisonBanner } from './LocationComparisonBanner';

interface LocationComparisonSectionProps {
  countryCode: string;
  countryName: string;
  themes: SpotlightTheme[];
}

export interface LocationTagProps {
  name: string;
  geocode: string;
}

export type LocationTagType = LocationTagProps[];

type NumberArray = number[];

const LocationComparisonSection: FunctionComponent<LocationComparisonSectionProps> = ({
  countryCode,
  countryName,
  themes
}) => {
  const [selectedLocations, setSelectedLocations] = useState<SpotlightLocation[]>([]);
  const [chartAndFilters, addChartAndFilters] = useState<NumberArray>([]);

  const onAddComparison = (): void => {
    addChartAndFilters([...chartAndFilters, 1]);
  };

  const renderAddComparisonComponents = (): ReactNode => {
    return chartAndFilters.map(
      (_item, index): ReactNode => {
        return (
          <PageSection key={index}>
            <Spotlight className="spotlight--full">
              <LocationFiltersAndCharts
                themes={themes}
                selectedLocations={selectedLocations}
                countryCode={countryCode}
              ></LocationFiltersAndCharts>
            </Spotlight>
          </PageSection>
        );
      }
    );
  };

  const onCompare = (locations: SpotlightLocation[] | any): void => {
    if (locations.length > 0) {
      setSelectedLocations(locations);
    }
  };

  return (
    <>
      <PageSection>
        <PageSectionHeading>Location Comparison</PageSectionHeading>
        <LocationComparisonBanner
          countryName={countryName}
          countryCode={countryCode}
          onCompare={onCompare}
        ></LocationComparisonBanner>
        <Spotlight className="spotlight--full">
          <LocationFiltersAndCharts
            themes={themes}
            selectedLocations={selectedLocations}
            countryCode={countryCode}
          ></LocationFiltersAndCharts>
        </Spotlight>
      </PageSection>
      {renderAddComparisonComponents()}
      <AddComparison onAddComparison={onAddComparison}></AddComparison>
    </>
  );
};

export { LocationComparisonSection };
