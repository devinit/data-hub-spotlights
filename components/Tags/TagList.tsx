import React, { FunctionComponent, Children, isValidElement } from 'react';
import { TagListItem } from './TagListItem';

export const TagList: FunctionComponent = ({ children }) => (
  <ul className="m-pills">
    {Children.map(children, (child) => (isValidElement(child) && child.type === TagListItem ? child : null))}
  </ul>
);
