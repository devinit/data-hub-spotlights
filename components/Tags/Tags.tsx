import React, { FunctionComponent, ReactNode } from 'react';
import { TagsList } from './TagsList';
import { TagsListItem } from './TagsListItem';
import { SpotlightLocation } from '../../utils';

interface TagsProps {
  updatedTags: SpotlightLocation[];
  onCloseTag: (tagName: string) => void;
}

const Tags: FunctionComponent<TagsProps> = ({ onCloseTag, updatedTags }) => {
  const onRemove = (label: string): void => {
    onCloseTag(label);
  };

  const renderLocationTagItems = (): ReactNode => {
    return updatedTags.map((tag: any, index: number) => {
      return <TagsListItem key={index} label={tag.name} onRemoveTag={onRemove} />;
    });
  };

  return (
    <>
      <TagsList>{renderLocationTagItems()}</TagsList>
    </>
  );
};

Tags.defaultProps = {
  updatedTags: []
};

export { Tags };
