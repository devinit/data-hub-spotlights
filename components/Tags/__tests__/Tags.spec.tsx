/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { Tags } from '../Tags';
import { LocationTagType } from '../../LocationComparisonSection';

const locations: LocationTagType = [
  { label: 'Masaka' },
  { label: 'Gulu' },
  { label: 'Arua' },
  { label: 'Koboko' },
  { label: 'Palisa' }
];
const onCloseTag = jest.fn();

describe('Tags', () => {
  test('It renders correctly', () => {
    const renderer = TestRenderer.create(<Tags onCloseTag={onCloseTag} updatedTags={locations} />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
