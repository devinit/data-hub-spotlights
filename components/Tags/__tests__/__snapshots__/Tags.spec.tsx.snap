// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags It renders correctly 1`] = `
<ul
  className="m-pills"
>
  <li
    className="m-pills__item"
  >
    <style
      jsx={true}
    >
      
        .m-pills__item {
          margin-bottom: 5px !important;
        }
      
    </style>
    <button
      onClick={[Function]}
      type="button"
    >
      <i
        aria-hidden="true"
        className="ico ico--20 ico-cross-slate"
        role="presentation"
      />
    </button>
    Masaka
  </li>
  <li
    className="m-pills__item"
  >
    <style
      jsx={true}
    >
      
        .m-pills__item {
          margin-bottom: 5px !important;
        }
      
    </style>
    <button
      onClick={[Function]}
      type="button"
    >
      <i
        aria-hidden="true"
        className="ico ico--20 ico-cross-slate"
        role="presentation"
      />
    </button>
    Gulu
  </li>
  <li
    className="m-pills__item"
  >
    <style
      jsx={true}
    >
      
        .m-pills__item {
          margin-bottom: 5px !important;
        }
      
    </style>
    <button
      onClick={[Function]}
      type="button"
    >
      <i
        aria-hidden="true"
        className="ico ico--20 ico-cross-slate"
        role="presentation"
      />
    </button>
    Arua
  </li>
  <li
    className="m-pills__item"
  >
    <style
      jsx={true}
    >
      
        .m-pills__item {
          margin-bottom: 5px !important;
        }
      
    </style>
    <button
      onClick={[Function]}
      type="button"
    >
      <i
        aria-hidden="true"
        className="ico ico--20 ico-cross-slate"
        role="presentation"
      />
    </button>
    Koboko
  </li>
  <li
    className="m-pills__item"
  >
    <style
      jsx={true}
    >
      
        .m-pills__item {
          margin-bottom: 5px !important;
        }
      
    </style>
    <button
      onClick={[Function]}
      type="button"
    >
      <i
        aria-hidden="true"
        className="ico ico--20 ico-cross-slate"
        role="presentation"
      />
    </button>
    Palisa
  </li>
</ul>
`;
